src/body.rs

```diff
@@ -150,6 +150,42 @@ impl<T: 'static + Send + serde::Serialize> IntoResponse for Json<T> {
     }
 }
 
+/// A wrapper for form encoded (application/x-www-form-urlencoded) (de)serialization of bodies.
+///
+/// This type is usable both as an extractor (argument to an endpoint) and as a response
+/// (return value from an endpoint), though returning a response with form data is uncommon
+/// and probably not good practice.
+pub struct Form<T>(pub T);
+
+impl<T: Send + serde::de::DeserializeOwned + 'static, S: 'static> Extract<S> for Form<T> {
+    // Note: cannot use `existential type` here due to ICE
+    type Fut = FutureObj<'static, Result<Self, Response>>;
+
+    fn extract(data: &mut S, req: &mut Request, params: &RouteMatch<'_>) -> Self::Fut {
+        let mut body = std::mem::replace(req.body_mut(), Body::empty());
+        FutureObj::new(Box::new(
+            async move {
+                let body = await!(body.read_to_vec()).map_err(mk_err)?;
+                let data: T = serde_qs::from_bytes(&body).map_err(mk_err)?;
+                Ok(Form(data))
+            },
+        ))
+    }
+}
+
+impl<T: 'static + Send + serde::Serialize> IntoResponse for Form<T> {
```

Comment:
I know there is another one in this file, but would it make sense to move this to the [Response.rs](https://github.com/rust-net-web/tide/blob/master/src/response.rs) file?

Reply:
I waffled on that and ended up keeping all code that was purely body-related here.

Reply:
Fair enough!